gtest/src/js/mod.rs

```diff
@@ -0,0 +1,176 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use log::debug;
+use std::path::Path;
+use std::process::Command;
+
+#[allow(dead_code)]
+pub enum MetaType {
+    InitInput,
+    InitOutput,
+    Input,
+    Output,
+}
+
+impl ToString for MetaType {
+    fn to_string(&self) -> String {
+        match self {
+            MetaType::InitInput => "init_input",
+            MetaType::InitOutput => "init_output",
+            MetaType::Input => "input",
+            MetaType::Output => "output",
+        }
+        .into()
+    }
+}
+
+pub fn gear_path() -> String {
+    let pwd = Command::new("pwd")
+        .output()
+        .expect("Unable to call pwd command");
+
+    let path = String::from_utf8(pwd.stdout).expect("Unable to parse pwd output bytes");
+    let path_parts: Vec<String> = path.split("/").map(|v| v.replace("\n", "")).collect();
```

Comment:
`PathBuf` has cross-platform `components` function for similar task. This code will not work on Windows.